3741/1/top_gems.rb

```diff
@@ -0,0 +1,31 @@
+require 'yaml'
+
+require_relative 'lib/yaml_reader'
+require_relative 'lib/gem_data_reader'
+require_relative 'lib/gems_view'
+require_relative 'lib/my_option_parser'
+
+# main run class
+class TopGems
+  def self.render_table
+    top_gems = TopGems.new
+    top_gems.show
+  end
+
+  FILE_NAME = 'gems.yml'.freeze
+
+  def initialize
+    @options = MyOptionParser.parse_options
+  end
+
+  def read_file
```

Comment:
you don't want to read file - you want to get gem_names 

change the name of the method